main.tf

```diff
@@ -106,133 +231,342 @@ data "aws_iam_policy_document" "secrets_access" {
+  role       = aws_iam_role.task_execution_role.name
+}
+# --- task role ---
+data "aws_iam_policy_document" "task_policy" {
+  version = "2012-10-17"
+  statement {
+    effect = "Allow"
+    principals {
+      identifiers = ["ecs-tasks.amazonaws.com"]
+      type        = "Service"
+    }
+    actions = ["sts:AssumeRole"]
+  }
+}
+resource "aws_iam_role" "task_role" {
+  name                 = "${var.app_name}-taskRole"
+  assume_role_policy   = data.aws_iam_policy_document.task_policy.json
+  permissions_boundary = var.role_permissions_boundary_arn
+  tags                 = var.tags
+}
+resource "aws_iam_role_policy_attachment" "task_policy_attach" {
+  count      = length(var.task_policies)
+  policy_arn = element(var.task_policies, count.index)
+  role       = aws_iam_role.task_role.name
+}
+resource "aws_iam_role_policy_attachment" "secret_task_policy_attach" {
+  count      = local.has_secrets ? 1 : 0
+  policy_arn = aws_iam_policy.secrets_access[0].arn
+  role       = aws_iam_role.task_role.name
+}
+# --- task definition ---
+resource "aws_ecs_task_definition" "task_def" {
+  container_definitions    = jsonencode(local.container_definitions)
+  family                   = "${var.app_name}-def"
+  cpu                      = var.task_cpu
+  memory                   = var.task_memory
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["FARGATE"]
+  execution_role_arn       = aws_iam_role.task_execution_role.arn
+  task_role_arn            = aws_iam_role.task_role.arn
+
+  tags = var.tags
+}
+
+# ==================== Fargate ====================
+resource "aws_ecs_cluster" "cluster" {
+  name = var.app_name
+  tags = var.tags
+}
+resource "aws_security_group" "fargate_service_sg" {
+  name        = "${var.app_name}-fargate-sg"
+  description = "Controls access to the Fargate Service"
+  vpc_id      = var.vpc_id
+
+  ingress {
+    from_port       = 0
+    to_port         = 65535
+    protocol        = "tcp"
+    security_groups = [aws_security_group.alb-sg.id]
+  }
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+  tags = var.tags
+}
+resource "aws_ecs_service" "service" {
+  name            = var.app_name
+  task_definition = aws_ecs_task_definition.task_def.arn
+  cluster         = aws_ecs_cluster.cluster.id
+  desired_count   = 1
+  launch_type     = "FARGATE"
+  deployment_controller {
+    type = "CODE_DEPLOY"
+  }
+  network_configuration {
+    subnets          = var.private_subnet_ids
+    security_groups  = [aws_security_group.fargate_service_sg.id]
+    assign_public_ip = true
+  }
+
+  load_balancer {
+    target_group_arn = aws_alb_target_group.blue.arn
+    container_name   = local.container_definitions[0].name
+    container_port   = var.image_port
+  }
+  //  load_balancer {
+  //    target_group_arn = aws_alb_target_group.green.arn
+  //    container_name = var.container_name
+  //    container_port = var.image_port
+  //  }
```

Comment:
?

Reply:
Removed. This was because I thought I had to define both target groups, but terraform/aws didn't like it since both target groups were not technically associated with an alb